throughput-client.c

```diff
@@ -0,0 +1,601 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <poll.h>
+#include <errno.h>
+#include <arpa/inet.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <net/if.h>
+#include <linux/sockios.h>
+#include <ifaddrs.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#define SERVER_PORT  42042
+#define CLIENT_PORT  0
+#define MAX_BUF_SIZE 1280	/* min IPv6 MTU, the actual data is smaller */
+#define MAX_TIMEOUT  5		/* in seconds */
+
+#define ENTRY(e, expect_result) { sizeof(e), e, expect_result }
+#define ENTRY_OK(e) ENTRY(e, true)
+#define ENTRY_FAIL(e) ENTRY(e, false)
+
+#define TYPE_SEQ_NUM 42
+
+struct header {
+	unsigned char type;
+	unsigned char len;
+	unsigned char value[0];
+} __packed;
+
+/* Next entry is 1280 bytes long which is the maximum length the IP stack
```

Comment:
I find such comments confusing. Which IP stack? If it's Zephyr stack, worth writing that, because e.g. Linux/other OS IP stack is certainly not subject to such restrictions.